### PR TITLE
Add a target to run all of the document tests together and make it work on Windows

### PR DESCRIPTION
**Pull request by sstamenova**
The DocExampleTest is meant as an example to be referenced on how to use the tools, so it makes sense for it to live with the documentation and be tested together. It also requires an installation of onnx (which the other doc test check-doc) does as well. It is the only part of the ALL target that requires an installation of onnx currently and makes it difficult to build ALL cleanly across platform.  See https://github.com/onnx/onnx-mlir/issues/561.

This change adds a new target to run both document tests together successfully (including on Windows) and makes the `ALL` target not require onnx installation.

Several things are needed to make this happen:

1. Move the document example test from under `test/unit/Runtime/DocExampleTest` to `docs/doc_example`
2. Add a new target `check-doc-example` which only executes the doc example test. This new target builds the test and executes it.
3. Add a new target `check-docs` which runs both the `check-doc` and `check-doc-example` targets allowing both to be run together
4. Mark the doc-example binary, so that it is not built as part of the `ALL` target. This means that none of the steps needed to build it (such as installing onnx, calling onnx-mlir, etc.) will be executed when the `ALL` target is built. These steps are executed when `check-doc-example` is run
5. Remove the explicit install of onnx through pip in the doc example test. The documentation states that this is needed for a variety of things to work correctly and all of the pipelines do it, so it is not needed in the cmake file.
6. Update the pipelines to use `check-docs` instead of `check-doc` - this will execute both tests together.
7. Several different scripts, as well as onnx-mlir, have been updated to work correctly on Windows

Notes:
- Temporarily (until there's a new clean target for the unit tests (see https://github.com/onnx/onnx-mlir/issues/835) the doc example test will be executed twice in the pipelines - once when `check-docs` is called, and once when `make test` is called. Since the test only takes a few seconds, I think this is acceptable.
- The new target is called `check-docs` and the document check target retains the name `check-doc`. I'm not super keen on either name, so if you have better suggestions, please let me know.
- This changes the name of the directory to `doc_example` to match other dirs in the same location. If you prefer a different naming convention, let me know
- The Windows-CI follows the convention of calling a script which contains the actual `cmake` steps. I've followed that here, but as far as I can tell the only reason to do so is to be able to reference the script in the documentation. If that's a valid reason, then I'll continue to do that for any future changes to the Windows-CI, but if it's not a requirement to have the scripts from the CI back the `README` files, I would prefer to not do that since it just complicates things.
- This change excludes the doc example binary from the `ALL` target, so calling `make test` without building the binary will now fail. I think this is fine since the goal is to have a `check-unittest` target to take over the rest of the unit tests and `make test` will no longer be necessary. If it's desired for `make test` to continue executing all available ctests, I can add a cmake option to build the test in the `ALL` target similar to llvm's `LLVM_BUILD_TESTS` 

Signed-off-by: Stella Stamenova <stilis@microsoft.com>

### CODE REVIEW

**Comment by caoimhinuibrian** [comment]
@jenkins-droid test this please

**Comment by AlexandreEichenberger** [comment]
@gongsu832 this changes looks good to me, since you have the most experience with our scripts, do you mind to give this patch your review? Thanks

**Comment by sstamenova** [comment]
@AlexandreEichenberger : I figured out what the issue was with the Windows-CI here - the machines where the tests run have multiple python installations. I fixed that, so the Windows-CI should be green with this change and execute the doc tests.

@gongsu832 : I'd particularly like your opinion on whether I should add a CMake variable to build the test executables.

**Comment by gongsu832** [comment]
@jenkins-droid test this please

File: docs/doc_example/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+  LINK_LIBS PRIVATE
+  OMRuntimeTestModelLibrary
+  cruntime
+  )
+set_output_directory(OMRuntimeTest BINARY_DIR ${CMAKE_CURRENT_BINARY_DIR} LIBRARY_DIR ${CMAKE_CURRENT_BINARY_DIR})
+
+add_test(NAME OMRuntimeTest COMMAND OMRuntimeTest WORKING_DIRECTORY ${CMAKE_CURRENT_BINARY_DIR})
+set_property(TARGET OMRuntimeTest PROPERTY FOLDER "Docs")
+set_tests_properties(OMRuntimeTest PROPERTIES LABELS doc-example)
+
```
Comment (gongsu832) [inline docs/doc_example/CMakeLists.txt:78]: The expectation is that `make test` should not fail after `make`. So yes as long as we still have `make test`, either we should make `OMRuntimeTest` work with `make test`, or we should remove it from `make test`.

**Comment by gongsu832** [comment]
Apart from the `OMRuntimeTest` issue, the rest look fine to me.

**Comment by AlexandreEichenberger** [comment]
@jenkins-droid test this please

**Comment by gongsu832** [comment]
@jenkins-droid test this please

File: docs/doc_example/CMakeLists.txt (hunk 1 above)
Comment (caoimhinuibrian) [reply docs/doc_example/CMakeLists.txt:82]: @sstamenova @gongsu832 Does something need to be done here?

Comment (sstamenova) [reply docs/doc_example/CMakeLists.txt:82]: I made the change to make `make test` continue to work as before by adding the `ONNX_MLIR_BUILD_TESTS` option. This allows `make test` to stay as it is unless `ONNX_MLIR_BUILD_TESTS` is set to `OFF` at configuration time.

**Review by gongsu832 — APPROVED** [review]
LGTM

**Comment by AlexandreEichenberger** [comment]
@sstamenova please look into the WindowCI error. Ready to merge once this is fixed. Thanks

**Comment by sstamenova** [comment]
> @sstamenova please look into the WindowCI error. Ready to merge once this is fixed. Thanks

@AlexandreEichenberger : The failure in the Windows CI seems like an intermittent infrastructure issue:

```
D:\a\2\llvm-project\llvm\lib\Transforms\InstCombine\InstCombineCasts.cpp : fatal error C1085: Cannot write compiler generated file: 'D:\a\2\llvm-project\build\lib\Transforms\InstCombine\CMakeFiles\LLVMInstCombine.dir\InstCombineCasts.cpp.obj': No space left on device
```

I'll merge with master again and I expect that the Windows CI will pass.

**Comment by gongsu832** [comment]
@jenkins-droid test this please

**Pull request merged**